test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -211,6 +213,116 @@ public void CreateDisplayMetadata_DisplayAttribute_DescriptionFromResources()
+        {
+            // Arrange
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory: null);
+
+            var display = new DisplayAttribute()
+            {
+                Name = "Model_Name",
+                Description = "Model_Description",
+                Prompt = "Model_Prompt"
+            };
+
+            var attributes = new Attribute[] { display };
+            var key = ModelMetadataIdentity.ForType(typeof(DataAnnotationsMetadataProviderTest));
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+
+            // Act
+            provider.CreateDisplayMetadata(context);
+
+            // Assert
+            Assert.Equal("Model_Name", context.DisplayMetadata.DisplayName());
+            Assert.Equal("Model_Description", context.DisplayMetadata.Description());
+            Assert.Equal("Model_Prompt", context.DisplayMetadata.Placeholder());
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_LocalizerMissingProperties()
+        {
+            // Arrange
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(s => s["Model_Name"])
+                .Returns(new LocalizedString("Model_Name", "Model_Name"));
+            stringLocalizer
+                .Setup(s => s["Model_Description"])
+                .Returns(new LocalizedString("Model_Description", "Model_Description"));
+            stringLocalizer
+                .Setup(s => s["Model_Prompt"])
+                .Returns(new LocalizedString("Model_Prompt", "Model_Prompt"));
+
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>(MockBehavior.Strict);
+            stringLocalizerFactory
+                .Setup(f => f.Create(It.IsAny<Type>()))
+                .Returns(stringLocalizer.Object);
+
+            var provider = new DataAnnotationsMetadataProvider(stringLocalizerFactory.Object);
+
+            var display = new DisplayAttribute()
+            {
+                Name = "Model_Name",
+                Description = "Model_Description",
+                Prompt = "Model_Prompt"
+            };
+
+            var attributes = new Attribute[] { display };
+            var key = ModelMetadataIdentity.ForType(typeof(DataAnnotationsMetadataProviderTest));
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+
+            // Act
+            provider.CreateDisplayMetadata(context);
+
+            // Assert
+            Assert.Equal("Model_Name", context.DisplayMetadata.DisplayName());
+            Assert.Equal("Model_Description", context.DisplayMetadata.Description());
+            Assert.Equal("Model_Prompt", context.DisplayMetadata.Placeholder());
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_DisplayAttribute_LocalizeProperties()
```

Comment:
Note this isn't really one test; should be broken into 3. We have too many tests like this already but, because they exist, this is just a suggestion.